### DISCUSSION TITLE
Query createOneGuild is required to return data, but found no record(s)

### DISCUSSION THREAD

**Discussion by Amgelo563** [prisma/prisma#13093] [Q&A] [answered]
Hello, I'm receiving an error when trying to create a simple record using Prisma with a MySQL provider.

`main.ts`
```ts
const { PrismaClient } = require("@prisma/client");

const prisma = new PrismaClient();

async function main() {
    console.log(await prisma.guild.create({ data: { id: 'some_id' } }));
}

main();
```

`schema.prisma`
```prisma
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "mysql"
  url      = env("DATABASE_URL")
}

model Guild {
  id    String  @id
}
```
Error
```
D:\Test\node_modules\@prisma\client\runtime\index.js:45633
        throw new PrismaClientUnknownRequestError(message, this.client._clientVersion);
              ^

PrismaClientUnknownRequestError:
Invalid `prisma.guild.create()` invocation in
D:\Software\Code\js\JsTestEnv\dist\index.js:47:55

  44 switch (_c.label) {
  45     case 0:
  46         _b = (_a = console).log;
→ 47         return [4 /*yield*/, prisma.guild.create(
  Query createOneGuild is required to return data, but found no record(s).
    at Object.request (D:\Test\node_modules\@prisma\client\runtime\index.js:45633:15)
    at async PrismaClient._request (D:\Test\node_modules\@prisma\client\runtime\index.js:46456:18) {
  clientVersion: '3.13.0'
}
```

This seems like a very simple issue but I couldn't find any answers or solutions. I'd be grateful if someone could point me into the right direction, thanks.

**Comment by janpio** [comment, marked as answer]
Was the entry actually created in the database?

**Reply by Amgelo563** [reply]
Upon checking with Prisma Studio, no there are no records created.

**Reply by janpio** [reply]
What version of MySQL are you using?
Does the exact same code as above cause this problem for you or is this a simplified example that might work just fine?

**Reply by Amgelo563** [reply]
The database is running MariaDB 10.x, currently asking host support which subversion exactly.
I ran into the error on production code, but the example provided works to reproduce it as well.

Edit:
MariaDB 10.3.29
MySQL 15.1

**Reply by janpio** [reply]
Ok, quick reproduction attempt with plain MySQL shows this schema+script working fine.

**Reply by Amgelo563** [reply]
What should I do to narrow down further the problem? Are there any common situations that could cause this? `npx prisma db push` seems to be working fine which is also odd.

**Reply by Amgelo563** [reply]
Created another database and now it seems to be working 🤔 odd problem; if someone runs into this, it seemed just to be a connection issue. Thanks for the help @janpio !

**Reply by janpio** [reply]
Super weird, but happy it works for you now. If it ever comes back, let us know and we will take another look to figure this out.

**Comment by acardosipendant** [comment]
Hello, I got this same error which in my case was caused because I specified `id` which was auto increment.
I was also running MySQL:
```prisma
generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["fullTextSearch", "fullTextIndex"]
}

datasource db {
  provider = "mysql"
  url      = env("DATABASE_URL")
}

model user {
  id     Int @id @default(autoincrement())
  name   String
}
```

I attempted to do
```ts
prisma.user.create({
  data: {
    id: 0,
    name: "User"
  }
})
```
But instead I needed to do:
```ts
prisma.user.create({
 data: {
    name: "User"
 }
})
``` 
EDIT: Changing it to `id: 1` also worked, rather than omitting `id` completely. Apparently it just can't handle `id: 0`

**Reply by sn0wst0rm** [reply]
@janpio I am currently facing the same problem with db seeding. I have a table that has some elements that must be at fixed and known indexes, and when I seed the table, I always get an error when trying to create the first element at id 0. I am using mariadb, but I don't think that's a database problem, because with a normal query I am able to create it.
It seems that Prisma itself can't handle a 0 index when creating. 

Also, even weirder, if I use another > 0 id and then use the update method to set the id to 0, it works, so it's a really strange behavior.

Do you know if a fix is possible?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
